internal/pkg/template/workload.go

```diff
@@ -205,7 +206,7 @@ type LogConfigOpts struct {
 	EnableMetadata *string
 	SecretOptions  map[string]Secret
 	ConfigFile     *string
-	Variables      map[string]string
+	Variables      map[string]Variable
```

Comment:
Was there a reason why you preferred using a `Variable` interface instead of making the `envVar` type public and getting rid of the interfaces?

Reply:
This is just so that the template YAML files would have a sort-of-stable interface to interact with. We can change the underlying implementation of the `envVar`'s method but the interface does not change. The benefit is marginal right now, but could be significant in the future if/when we let users bring their own go template. Plus this is largely inspired by https://github.com/aws/copilot-cli/blob/2d3a3f8036941d4f366fc0888b03c7f8eb573f3b/internal/pkg/template/workload.go#L255